### ISSUE TITLE
info: LFW 99.80 new approach

### ISSUE THREAD

**Issue by MyraBaba** [davisking/dlib#1244]
I want to share with you guys : https://github.com/deepinsight/insightface#deep-face-recognition

ArcFace, was initially described in an arXiv technical report. By using this repository, you can simply achieve LFW 99.80%+ and Megaface 98%+ by a single model.  https://arxiv.org/abs/1801.07698 

may be we can utilize with same approach in c++11/dlib  feature extraction.

I would like hear comments.

Best

**Comment by cftang0827** [comment]
According to [the training code](http://dlib.net/dnn_metric_learning_on_images_ex.cpp.html), dlib face recognition model use metric learning to estimate the L2 distance of two faces, according to the repository you post, the main difference is the loss function, right?

If there's any misunderstanding, I am sorry for that :)

**Comment by MyraBaba** [comment]
Looks like basically loss function. According to the paper: 

> To enhance the discriminative power of the Softmax loss, multiplicative angular margin and additive cosine margin incorporate angular margin and cosine margin into the loss functions, respectively. In this paper, we propose a novel supervisor signal, additive angular margin (ArcFace), which has a better geometrical interpretation than supervision signals proposed so far. Specifically, the proposed ArcFace cos(θ+m) directly maximise decision boundary in angular (arc) space based on the L2 normalised weights and features. Compared to multiplicative angular margin cos(mθ) and additive cosine margin cosθ−m, ArcFace can obtain more discriminative deep features.

I belive this could be increase the recognition accuracy.

But I am not yet so comfortable to implement this  in c++
